Remove `track-kind.html` in favor of WPT coverage

Ahmad-S792 opened the pull request:
#### 95006eac3dd163851d8ca029c3752c11177b1704
<pre>
Remove `track-kind.html` in favor of WPT coverage
<a href="https://bugs.webkit.org/show_bug.cgi?id=298478">https://bugs.webkit.org/show_bug.cgi?id=298478</a>
<a href="https://rdar.apple.com/159973501">rdar://159973501</a>

Reviewed by Tim Nguyen and Eric Carlson.

This patch removes &apos;track-kind.html&apos;, which is covered by WPT test cases
below:

<a href="https://wpt.fyi/results/html/semantics/embedded-content/media-elements?label=master&amp">https://wpt.fyi/results/html/semantics/embedded-content/media-elements?label=master&amp</a>;label=experimental&amp;aligned&amp;q=kind.html

Same was done in Blink here: <a href="https://chromium.googlesource.com/chromium/src.git/+/7284e5e7452bfeb539b7d1f3c698d8b967371d19">https://chromium.googlesource.com/chromium/src.git/+/7284e5e7452bfeb539b7d1f3c698d8b967371d19</a>

* LayoutTests/TestExpectations:
* LayoutTests/media/track/track-kind-expected.txt: Removed.
* LayoutTests/media/track/track-kind.html: Removed.

Canonical link: <a href="https://commits.webkit.org/299653@main">https://commits.webkit.org/299653@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/24d138d640e2587cdeee4506650d54b50f6162a3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119717 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39409 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126010 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71789 "Built successfully") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40105 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47987 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/90923 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/60218 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122669 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32013 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107345 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71450 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/31046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25450 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69650 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101483 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25641 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128963 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46637 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/35331 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/99522 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47002 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103537 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99366 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25235 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44798 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22821 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/43202 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46499 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45965 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49314 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47651 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/24d138d640e2587cdeee4506650d54b50f6162a3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50392

Ahmad-S792 committed:
Remove `track-kind.html` in favor of WPT coverage
https://bugs.webkit.org/show_bug.cgi?id=298478
rdar://159973501

Reviewed by Tim Nguyen and Eric Carlson.

This patch removes 'track-kind.html', which is covered by WPT test cases
below:

https://wpt.fyi/results/html/semantics/embedded-content/media-elements?label=master&label=experimental&aligned&q=kind.html

Same was done in Blink here: https://chromium.googlesource.com/chromium/src.git/+/7284e5e7452bfeb539b7d1f3c698d8b967371d19

* LayoutTests/TestExpectations:
* LayoutTests/media/track/track-kind-expected.txt: Removed.
* LayoutTests/media/track/track-kind.html: Removed.

Canonical link: https://commits.webkit.org/299653@main
